static/js/app.js

```diff
@@ -3,106 +3,158 @@
+  //array to hold everything that needs to be highlighted in doc
+  let highlights = [];
 
-    //loop through the stored annotations and append to the array as a dictionary
-    currentDocument.annotations.forEach(function(annotation) {
-        let single_highlight = {};
+  //loop through the stored annotations and append to the array as a dictionary
+  currentDocument.annotations.forEach(function (annotation) {
+    let single_highlight = {};
 
-        single_highlight.highlight = [annotation.range.startPosition, annotation.range.endPosition];
-        single_highlight.className = annotation.highlighter_color;
+    single_highlight.highlight = [annotation.range.startPosition, annotation.range.endPosition];
+    single_highlight.className = "label_" + annotation.label;
 
-        highlights.push(single_highlight);
-    });
+    highlights.push(single_highlight);
+  });
 
-    //then highlight based on that array
-    $('textarea').highlightWithinTextarea({
-        highlight: highlights
-    });
+  //then highlight based on that array
+  $('textarea').highlightWithinTextarea({
+    highlight: highlights
+  });
 };
 
 //change the document's label context
-$('input[type=radio]').change(function () {
-    currentLabel = this.value;
+$('.label').on('click', function () {
+  console.log(this.getAttribute('value'));                        //remove when done
+  console.log($(this).css('background-color'));                   //remove when done
+
+  currentLabel = this.getAttribute('value');
+  $('.label').attr('id', '');
+  $(this).attr('id', 'label-selected');
 });
 
-//change document's highlighter color context
-$('.highlight_color').click(function () {
-    if (currentHighlighterColor) {
-        $("#" + currentHighlighterColor).css('border-width', "thin");
-    }
-    currentHighlighterColor = this.id;
-    $(this).css('border-width', "medium");
-    renderTextareaHighlights();
+//invoke colorpicker on icon click
+$('.colorChange').on('click', function () {
+  console.log("click") //remove when done
+  $(this).parent().children('input').click();
+});
+
+//change label color on 
+$('.colorChangePicker').on('change', function () {
+  console.log(this.value);                                        //remove when done
+  console.log($(this).parent()[0].getAttribute('value'));         //remove when done
+  var parent = $(this).parent()[0];
+  $(this).parent().css('background-color', this.value);
+  $('#' + $(this).parent()[0].getAttribute('value') + '-style').html('.hwt-content .label_' + $(this).parent()[0].getAttribute('value') + ' {background-color: ' + this.value + ';}');
+});
+
+//update height on window resize
+$(window).on('resize', function () {
+  $('#doc-view').css('height', 'auto');
+  $('#doc-view').height($('#doc-view').prop('scrollHeight'));
 });
+//calling this twice fixes an issue with height
+$('#doc-view').height($('#doc-view').prop('scrollHeight'));
+$('#doc-view').height($('#doc-view').prop('scrollHeight'));
```

Comment:
Can also be put in a variable to improve performance

Reply:
updated for better performance + less redundancy x2